app/src/main/java/com/github/polypoly/app/ui/menu/lobby/GameLobbyActivity.kt

```diff
@@ -70,162 +83,623 @@ class GameLobbyActivity : ComponentActivity() {
+        Row {
+            Text(
+                text = getString(R.string.game_lobby_more_players_prefix),
+                style = MaterialTheme.typography.body1,
+                modifier = Modifier
+                    .testTag("game_lobby_start_game_button_more_players_needed_prefix")
+            )
+            Text(
+                text = getString(R.string.game_lobby_more_players_count, morePlayersNeeded.toString()),
+                style = MaterialTheme.typography.body1.copy(
+                    color = MaterialTheme.colors.primary,
+                    fontWeight = FontWeight.Bold
+                ),
+                modifier = Modifier
+                    .testTag("game_lobby_start_game_button_more_players_needed_count")
+            )
+            Text(
+                text = getString(R.string.game_lobby_more_players_suffix),
+                style = MaterialTheme.typography.body1,
+                modifier = Modifier
+                    .testTag("game_lobby_start_game_button_more_players_needed_suffix")
+            )
+        }
+    }
+
+    /**
+     * Display the current player count in relation to the maximum player count.
+     * @param players the list of players in the game lobby.
+     * @param maxPlayers the maximum number of players allowed in the game lobby.
+     */
+    @Composable
+    fun CurrentPlayerCountDisplay(players: List<User>, maxPlayers: Int) {
+        Row {
+            Text(
+                text = getString(R.string.game_lobby_players_title),
+                style = MaterialTheme.typography.body1,
+                modifier = Modifier
+                    .testTag("game_lobby_start_game_button_current_player_count_title")
+            )
+            Text(
+                text =  getString(R.string.game_lobby_players_count, players.size.toString(), maxPlayers.toString()),
+                style = MaterialTheme.typography.body1.copy(
+                    color = MaterialTheme.colors.primary
+                ),
+                modifier = Modifier
+                    .testTag("game_lobby_start_game_button_current_player_count")
+            )
+        }
+    }
+
+    /**
+     * Launch the game activity and finish this activity. Only available to the admin.
+     * will set the game lobby as started in the database.
+     */
     private fun launchGameActivity(packageContext: Context) {
         val completedLobby = gameLobbyWaitingModel.getGameLobby().value
         if (completedLobby != null) {
-            GameRepository.game = Game.launchFromPendingGame(completedLobby)
-            val gameIntent = Intent(packageContext, GameActivity::class.java)
-            startActivity(gameIntent)
-            finish()
+
+            val newGameLobby = GameLobby(
+                completedLobby.admin,
+                completedLobby.rules,
+                completedLobby.name,
+                completedLobby.code,
+                completedLobby.private,
+                true
+            )
```

Comment:
There is a function `start` that you could adapt to both set its attribute `started` to true and return the generated GameLobby. It would also have the benefit of checking again that the lobby can start.

Reply:
looks like a nice idea. I will mark a TODO to look into it in a further sprint! (this would require to adapt tests for Game, Player, GameLobby etc... That is why I don't want to work about it in this PR that is already big enough)